From 1744e433a16fb39a55c4c771b8ecb02246aa5b99 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 15 Mar 2025 09:40:06 -0700
Subject: [PATCH] acinclude.m4: Add parameter names in prototype for g().
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This allows it to compile with older gcc e.g. gcc-10
which does not have allow parameter name omission, it results
in

a.c: In function ‘g’:
a.c:3:8: error: parameter name omitted
    3 | void g(int,t1 const*,t1,t2,t1 const*,int){}
      |        ^~~

this was added to gcc via [1] thats why it is supported in
newer gcc.

Adding the parameter names make it compatible with
old and new gcc

[1] https://gcc.gnu.org/pipermail/gcc-cvs/2020-October/336068.html

Upstream-Status: Submitted [https://gmplib.org/list-archives/gmp-devel/2025-March/006294.html]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
ChangeLog

2025-03-15  Khem Raj <raj.khem@gmail.com>

   * acinclude.m4: Add parameter names to function prototype.

 acinclude.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/acinclude.m4 b/acinclude.m4
index 4fca12de2..b9d1eacfe 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -609,7 +609,7 @@ GMP_PROG_CC_WORKS_PART([$1], [long long reliability test 1],
 
 #if defined (__GNUC__) && ! defined (__cplusplus)
 typedef unsigned long long t1;typedef t1*t2;
-void g(int,t1 const*,t1,t2,t1 const*,int){}
+void g(int a,t1 const* b,t1 c,t2 d,t1 const* e,int f){}
 void h(){}
 static __inline__ t1 e(t2 rp,t2 up,int n,t1 v0)
 {t1 c,x,r;int i;if(v0){c=1;for(i=1;i<n;i++){x=up[i];r=x+1;rp[i]=r;}}return c;}
